presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java

```diff
@@ -160,20 +160,20 @@ public HivePartitionResult getPartitions(SemiTransactionalHiveMetastore metastor
 
     private static TupleDomain<HiveColumnHandle> toCompactTupleDomain(TupleDomain<ColumnHandle> effectivePredicate, int threshold)
     {
-        checkArgument(effectivePredicate.getDomains().isPresent());
-
         ImmutableMap.Builder<HiveColumnHandle, Domain> builder = ImmutableMap.builder();
-        for (Map.Entry<ColumnHandle, Domain> entry : effectivePredicate.getDomains().get().entrySet()) {
-            HiveColumnHandle hiveColumnHandle = (HiveColumnHandle) entry.getKey();
-
-            ValueSet values = entry.getValue().getValues();
-            ValueSet compactValueSet = values.getValuesProcessor().<Optional<ValueSet>>transform(
-                    ranges -> ranges.getRangeCount() > threshold ? Optional.of(ValueSet.ofRanges(ranges.getSpan())) : Optional.empty(),
-                    discreteValues -> discreteValues.getValues().size() > threshold ? Optional.of(ValueSet.all(values.getType())) : Optional.empty(),
-                    allOrNone -> Optional.empty())
-                    .orElse(values);
-            builder.put(hiveColumnHandle, Domain.create(compactValueSet, entry.getValue().isNullAllowed()));
-        }
+        effectivePredicate.getDomains().ifPresent(domains -> {
```

Comment:
Without this one of the tests fails (can't remeber which one, and the tests now fail in this branch, so can't find it easily...). OTOH @sopel39 suggested that this might actually be working around another bug that only surfaced after adding cost calculation for each query (which this commit does). @sopel39 also claims the check state should stay here (right?). We'll investigate this after the tests pass again.